### ISSUE TITLE
Request: multiple cameras at once

### ISSUE THREAD

**Issue by rektide** [Genymobile/scrcpy#4451]
- [x] I have checked that a similar [feature request](https://github.com/Genymobile/scrcpy/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22) does not already exist.

**Is your feature request related to a problem? Please describe.**
This is a feature request!

**Describe the solution you'd like**
Ideally I could run multiple copies of `scrcpy` with different `--camera-id`'s specified and get multiple different cameras streaming at once into different windows.

**Describe alternatives you've considered**
One alternative that likely would be acceptable for me is to stream multiple cameras as multiple different windows. This is not-optimal for me, as I can't turn cameras on and off on the fly; both cameras need to be configured ahead of time and stay on. But it would be better than what I have now.

One thought I had but wasn't able to follow up on is maybe it's possible to create multiple different multiple serial links to my phone and that maybe that might help. I don't know `adb` very well nor it's technical limitations, but I poked around for a bit & I don't think I got anywhere in terms of setting up multiple connections. One thing I tried here was seeing if I could use -P port to maybe launch multiple adb servers at once and have them each pair, but it was hard to tell what was really happening.

Nightmare scary mode (to me)_might be trying to make one `scrcpy` somehow blit multiple cameras or screens together in some way. That seems like it would be hard to configure well.

**Additional context**
I am using adb-over-wifi serial to talk to a Samsung S22, with Android 13. Right now I'm on a windows box. This is what 

I'm trying to run scrcpy twice, as so:

```sh-session
$ scrcpy -s adb-R5CRC3MLEFK-Po0SPe._adb-tls-connect._tcp. --no-audio --video-bit-rate=2M --video-source=camera --camera-size=960x720 --camera-id=0 --window-title=phonecam-back
$ scrcpy -s adb-R5CRC3MLEFK-Po0SPe._adb-tls-connect._tcp. --no-audio --video-bit-rate=2M --video-source=camera --camera-size=960x720 --camera-id=1 --window-title=phonecam-front
```

When I start the second scrcpy, the first instance dies as so, closing the window, and then the new scrcpy window starts up with the desired video camera (as normal, without trouble):

```
[server] WARN: Camera disconnected
[server] WARN: Camera capture failed: frame 121
[server] WARN: Camera capture failed: frame 122
[server] WARN: Camera capture failed: frame 123
[server] WARN: Camera capture failed: frame 124
[server] WARN: Camera capture failed: frame 125
[server] WARN: Camera capture failed: frame 120
WARN: Device disconnected
```

One additional bit of context, I can run either the front or back camera while also capturing the screen. Ex of that invocation:

```sh-session
$ scrcpy -s adb-R5CRC3MLEFK-Po0SPe._adb-tls-connect._tcp.  --no-audio --video-bit-rate=3M --window-title=phone 
```

I've also played with various settings some, to no effect. I've tried running one stream with `--audio-source=mic` and one with `--audio-source=output`. I've tried with one `--no-audio` and the other a source. I've tried using some other resolutions (640x480). I've tried `--video-codec=h264` and each of the 4 different `--video-encoder` options shown for h264 in `--list-encoders` (full list: `OMX.google.h264.encoder c2.qti.avc.encoder OMX.qcom.video.encoder.avc c2.android.avc.encoder`). One thing that I thought was promising was to specify a `--port=28183:28199` non-default port for the second camera, but that didn't help. So far none of these have made a difference that I've seen.

**Comment by rom1v** [comment]
> When I start the second scrcpy, the first instance dies as so, closing the window, and then the new scrcpy window starts up with the desired video camera (as normal, without trouble)

Yes, it is the (physical, I guess) limitation on the Android device. As soon as we start a new camera capture, ne old one get disconnected.

There's probably nothing scrcpy can do.

**Comment by yume-chan** [comment]
https://developer.android.com/reference/android/hardware/camera2/CameraManager#getConcurrentCameraIds()

> The devices in these combinations can be concurrently configured by the same client camera application. Using these camera devices concurrently by two different applications is not guaranteed to be supported, however.
> 
> For concurrent operation, in chronological order :
> 
> * Applications must first close any open cameras that have sessions configured, using [CameraDevice#close](https://developer.android.com/reference/android/hardware/camera2/CameraDevice#close()).
> * All camera devices intended to be operated concurrently, must be opened using [openCamera(String, StateCallback, Handler)](https://developer.android.com/reference/android/hardware/camera2/CameraManager#openCamera(java.lang.String,%20android.hardware.camera2.CameraDevice.StateCallback,%20android.os.Handler)), before configuring sessions on any of the camera devices.

1. Maybe one Scrcpy instance can open multiple cameras. This requires a huge refactoring to allow one Scrcpy server to accept multiple client connections.
2. Opening a new camera requires closing and re-opening other already opened cameras.
3. Depends on the device, some camera combinations might not work.

**Comment by TheBug233** [comment]
My phone natively supports simultaneous shooting with multiple cameras (two rear cameras and one front camera). Additionally, in the DSU Loader system, it's possible to mirror two cameras simultaneously in Scrcpy. However, it's unfortunate that DSU Loader cannot recognize the wide-angle camera on my phone.

So, is this a software limitation of the phone's system? Can it be bypassed with root access or a Magisk module?
